core/src/main/scala/org/apache/spark/deploy/JsonProtocol.scala

```diff
@@ -54,7 +55,7 @@ private[deploy] object JsonProtocol {
 
   def writeApplicationDescription(obj: ApplicationDescription): JObject = {
```

Comment:
IMO, we need to add the function descriptions for these public functions and provide a clear definition for each field. How about adding a few JIRAs and ask the open source community to do it?